src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -1006,9 +1006,10 @@ public override int GetHashCode()
         [Pure]
         public override bool Equals(object obj)
         {
-            if (obj is ImmutableArray<T>)
+            IImmutableArray other = obj as IImmutableArray;
```

Comment:
I removed the check for `ImmutableArray<T>` altogether since `obj` is already boxed on entering this method.